ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlAbstractDecoder.kt

```diff
@@ -26,8 +26,8 @@ public abstract class TomlAbstractDecoder : AbstractDecoder() {
     override fun decodeByte(): Byte = decodePrimitiveType()
     override fun decodeShort(): Short = decodePrimitiveType()
     override fun decodeInt(): Int = decodePrimitiveType()
-    override fun decodeFloat(): Float = decodePrimitiveType()
-    override fun decodeChar(): Char = decodePrimitiveType()
+    override fun decodeFloat(): Float = invalidType("Float", "Double")
```

Comment:
Let's split this logic into several PRs, same with Char.

Actually Char is even more discussible, because there is no syntax in Toml for char literals, only for Strings...

Reply:
Yeah, I copy and pasted Char in, but in retrospect I don't think it's clear how it should be handled. Is Char a number? Should `101` be treated as ascii and be decoded to `A`? (No, probably not.) What about decoding `a` to `val c: Char`?